File: csa-official-backend/src/main/java/com/csa/official/modules/sys/entity/Dept.java
package com.csa.official.modules.sys.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("sys_dept")
public class Dept implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String intro;

    // 当前正部长的用户ID
    private Long leaderId;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: csa-official-frontend/src/services/dept.ts
import api from "@/lib/axios"

export interface DeptItem {
  id: number
  name: string
  intro: string | null
  leaderId: number | null
  createTime: string | null
  updateTime: string | null
}

export const deptService = {
  list: () => api.get<DeptItem[], DeptItem[]>("/api/sys/dept/list"),

  appoint: (deptId: number, userId: number) =>
    api.post<string, string>("/api/sys/dept/appoint", {
      deptId,
      userId,
    }),
}
